### DISCUSSION TITLE
Tern as Docker image

### DISCUSSION THREAD

**Discussion by ThiefaineM** [tern-tools/tern#1138] [Q&A] [answered]
Hello,

Is Tern distributed as a Docker image is still in project ?
Or is it stopped in favor of  Kubernetes ?

Thanks in advance :)

**Comment by rnjudge** [comment, marked as answer]
Hi @ThiefaineM -- Tern does not have its own Docker image that you can pull from an existing container registry. However, you can build your own Docker image and run Tern using one of the [Dockerfiles](https://github.com/tern-tools/tern/tree/main/docker) provided in the repo. We are currently focused on running Tern as a service in Kubernetes in favor of distributing a separate Docker image (see the [API repository](https://github.com/tern-tools/tern-rest-api) as well as the [issue](https://github.com/tern-tools/tern/issues/1112)). Tern currently runs as a Kubernetes job using these [directions](https://github.com/tern-tools/tern/blob/f468c1d72160de39a4805370a38db91d3ed9d1a2/README.md#creating-a-kubernetes-job).

Note that the Dockerfiles in the repo I linked to will build Tern from the latest pypi release. If you want to install Tern from the tip of master, first clone Tern and run `python setup.py bdist_wheel`. Then use this [Dockerfile](https://github.com/tern-tools/tern/blob/main/ci/Dockerfile) to build the container.
